transmogrifier/sources/xml/oaidc.py

```diff
@@ -36,45 +36,34 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
         # citation: uses fallback get_citation() method
 
         # content_type
-        fields["content_type"] = [self.source]
+        fields["content_type"] = self.get_content_type()
 
         # contents: not set in this transformation
 
         # contributors
-        for creator in [c for c in xml.find_all("dc:creator") if c.string]:
-            fields.setdefault("contributors", []).append(
-                timdex.Contributor(
-                    value=str(creator.string),
-                    kind="Creator",
-                )
-            )
+        fields["contributors"] = self.get_contributors(source_record) or None
 
         # dates
-        fields["dates"] = self.get_dates(source_record_id, xml) or None
+        fields["dates"] = self.get_dates(source_record, source_record_id) or None
 
         # edition: not set in this transformation
 
         # file_formats: not set in this transformation
 
         # format
-        fields["format"] = "electronic resource"
+        fields["format"] = self.get_format()
 
         # funding_information: not set in this transformation
 
         # holdings: not set in this transformation
 
         # identifiers
-        fields.setdefault("identifiers", []).append(
-            timdex.Identifier(
-                value=str(xml.header.identifier.string),
-                kind="OAI-PMH",
-            )
-        )
+        fields["identifiers"] = self.get_identifiers(source_record) or None
 
         # languages: not set in this transformation
 
         # links
-        fields["links"] = self.get_links(source_record_id, xml) or None
+        fields["links"] = self.get_links(source_record, source_record_id) or None
```

Comment:
Do we still need to pass `source_record_id`, if the `source_record` contains it?

And/or, could the method `get_links()` just call the _other_ method `get_source_record_id()` again if not?  

Feeling like maybe this is touched on in other comments.  Full disclaimer: I wrote these transforms fairly early on, and recall they had requirements that deviated from previous Transmog sources (as I think you've commented on as well @jonavellecuerdo, like the identifier vs source link).  So it feels possible that now is a good time to revisit some of those awkward edges.  Silver lining: these are not yet live, so I think we're fairly free to modify them somewhat heavily if needed.